nodes/node_balancer/cmd/data.go

```diff
@@ -17,6 +17,23 @@ type NodeStatusResponse struct {
 	CurrentBlock uint64 `json:"current_block"`
 }
 
+// Node - which one node client worked with
+// LastCallTs - timestamp from last call
+type ClientNode struct {
+	Blockchain string
+	Node       *Node
+	LastCallTs int64
+}
+
+type Client struct {
+	IP          string
+	ClientNodes []ClientNode
+}
+
+type ClientPool struct {
+	Clients []*Client
```

Comment:
I would instead expect:

```
type ClientPool struct {
    Clients map[string]*Node
}
```

Do you have a use case for multiple client nodes per client?

Instead of putting blockchain in the `ClientNode` struct, I would recommend creating two client pools variables on the server - `ethereumClientPool` and `polygonClientPool`. I think that generalizes better.

Reply:
Made this as map[string]*Node, but with different polygon and ethereum clientPools it brings a lot of code duplication. Because I am using structure:
```
func (cpool *ClientPool) AddClientNode(id, blockchain string, node *Node) {}
```
And can't inherit from `ethereumClientPool` and `polygonClientPool` same time, so it will require code duplication